File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if len(nums) == 0 : return 0
        cur =2
        for ni in range(2,len(nums)):
            if nums[ni] == nums[cur-1] == nums[cur-2]:
                continue
            nums[cur] = nums[ni]
            cur += 1
        for _ in range(len(nums)-cur):
            nums.pop()
        return len(nums)

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 1) return 1;
        int left = 0;
        int i = 0;
        for(int right = 0; right<nums.length;right++){
            if(nums[left] == nums[right]){
                if(right - left <2){
                    nums[i] = nums[right];
                    i+=1;
                }
            }else{
                left = right;
                nums[i] = nums[right];
                i++;
            }
        }
        return i;
    }
}